File: 3005-count-elements-with-maximum-frequency/3005-count-elements-with-maximum-frequency.java
class Solution {
    public int maxFrequencyElements(int[] nums) {
        int ans =0;
        int maxi =0;
        HashMap<Integer,Integer> map = new HashMap<>();
        for(int x: nums){
            map.put(x,map.getOrDefault(x,0)+1);
            maxi = Math.max(maxi,map.get(x));
        }
        for(int x: map.keySet()){
            if(map.get(x) == maxi){
                ans += map.get(x); 
            }
        }
        return ans;
        
    }
}

File: 3005-count-elements-with-maximum-frequency/3005-count-elements-with-maximum-frequency.cpp
class Solution {
public:
    int maxFrequencyElements(vector<int>& nums) {
        int ans =0;
        unordered_map<int,int> map;
        int maxi =0;
        for(int x: nums){
            map[x]++;
            maxi = max(maxi,map[x]);
        }
        set<int> uni(nums.begin(),nums.end());
        for(int x: uni){
            if(map[x] == maxi){
                ans += map[x];
            }
        }
        return ans;
    }
};